src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/DockerProcessor.kt

```diff
@@ -27,9 +27,14 @@ import java.nio.file.Paths
 const val PDF_EXTENSION = ".pdf"
 
 class DockerProcessor {
-    private val docker: DockerClient = DefaultDockerClient.fromEnv().build()
+    private var docker: DockerClient? = null
 
     fun compileRmdToPdf(rootFile: File): File {
+        if (docker == null) {
```

Comment:
Pass docker client to the constructor of this class and remove these null checks, mutability of `docker` field and finally all that injection magic.

Reply:
Done